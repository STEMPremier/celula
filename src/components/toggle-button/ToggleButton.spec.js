/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { ToggleButtonGroup, ToggleButton } from '.';

describe('<ToggleButton', () => {
  describe('render()', () => {
    test('render the ToggleButtonGroup and Toggle components', () => {
      const wrapper = shallow(
        <ToggleButtonGroup
          handleChange={() => {}}
          name="formTest"
          selectedValue="follow"
        >
          <ToggleButton icon="export" value="export" />
          <ToggleButton icon="follow" value="follow" />
          <ToggleButton icon="message" value="message" />
        </ToggleButtonGroup>,
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
/* eslint-enable react/jsx-filename-extension */
