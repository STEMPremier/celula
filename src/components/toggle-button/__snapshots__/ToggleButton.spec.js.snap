// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButton render() render the ToggleButtonGroup and Toggle components 1`] = `
<FormControlGroup
  childProps={
    Object {
      "color": "primary",
      "size": "large",
    }
  }
  className="ce-toggle-button-group ce-toggle-button-group--large ce-toggle-button-group--primary"
  disabled={false}
  errorClass=""
  errorMsg=""
  formId=""
  handleChange={[Function]}
  label=""
  name="formTest"
  selectedValues={
    Array [
      "follow",
    ]
  }
>
  <ToggleButton
    checked={false}
    className=""
    disabled={false}
    icon="export"
    isA="radio"
    size="large"
    value="export"
  />
  <ToggleButton
    checked={false}
    className=""
    disabled={false}
    icon="follow"
    isA="radio"
    size="large"
    value="follow"
  />
  <ToggleButton
    checked={false}
    className=""
    disabled={false}
    icon="message"
    isA="radio"
    size="large"
    value="message"
  />
</FormControlGroup>
`;
